actors/ppo_atari_pz.py

```diff
@@ -0,0 +1,1034 @@
+            # Store data
+            if (i + 1) % batch_size == 0:
+                data_loader.append(output_batches)
+
+                # Reset
+                count = 0
+                output_batches = torch.zeros((batch_size, *obs_shape), dtype=self._dtype).to(self._device)
+            else:
+                count += 1
+                if i == num_data - 1:
+                    data_loader.append(output_batches[:count])
+
+        return data_loader
+
+    def normalize_rewards(self, rewards: list, model: PPOModel) -> list:
+        """Normalize the rewards"""
+        normalized_reward = []
+        for rew in rewards:
+            normalized_reward.append(rew / (model.reward_normalizaiton.var + 1e-8) ** 0.5)
+            self.returns = self.returns * self._cfg.discount_factor + rew
+            model.reward_normalization.update(self.returns)
+
+        return normalized_reward
+
+    def compute_gae(
+        self, rewards: torch.Tensor, values: torch.Tensor, dones: torch.Tensor, gamma: float = 0.99, lam: float = 0.95
+    ) -> torch.Tensor:
+        """Compute Generalized Advantage Estimation. See equations 11 & 12 in
+        https://arxiv.org/pdf/1707.06347.pdf
+        """
+
+        advs = []
+        with torch.no_grad():
+            gae = 0.0
+            # dones = torch.cat((dones, torch.zeros(1, 1).to(self._device)), dim=0)
+            for i in reversed(range(len(rewards))):
+                delta = rewards[i] + gamma * values[i + 1] * (1 - dones[i + 1]) - values[i]
+                gae = delta + gamma * lam * (1 - dones[i + 1]) * gae
+                advs.append(gae)
+        advs.reverse()
+        return torch.vstack(advs)
+
+
+class PPOSelfTraining(BasePPOTraining):
+    """Train PPO agent"""
+
+    async def impl(self, run_session: RunSession) -> dict:
+        """Train and publish the model"""
+        model_id = f"{run_session.run_id}_model"
+
+        # Initalize model
+        self.model.model_id = model_id
+        _, version_info = await run_session.model_registry.publish_initial_version(self.model)
+        self.model.network.to(self._device)
+
+        run_session.log_params(
+            self._cfg,
+            model_id=model_id,
+            environment_implementation=self._environment_specs.implementation,
+            hill_type="none",
+        )
+
+        # Helper function to create a trial configuration
+        def create_trial_params(trial_idx: int, iter_idx: int):
+            agent_actor_params = cogment.ActorParameters(
+                cog_settings,
+                name="player",
+                class_name=PLAYER_ACTOR_CLASS,
+                implementation="actors.ppo_atari_pz.PPOActor",
+                config=AgentConfig(
+                    run_id=run_session.run_id,
+                    environment_specs=self._environment_specs.serialize(),
+                    model_id=model_id,
+                    model_version=version_info["version_number"],
+                    seed=self._cfg.seed,
```

Comment:
This uses the same seed for every trials, I think it'd be better to do the same kind of stuff that you did for the environment (you can also just use the rng of the training to pick a random seed).